.github/workflows/e2e-tests-main-devnet.yml

```diff
@@ -510,6 +665,10 @@ jobs:
       run-e2e-ban-manual,
       run-e2e-version-upgrade,
       run-e2e-failing-version-upgrade,
+      run-e2e-version-upgrade-catchup-one-node-late,
+      run-e2e-version-upgrade-catchup-two-nodes-late,
+      run-e2e-version-upgrade-early-catchup-one-node-late,
+      run-e2e-version-upgrade-early-catchup-two-nodes-late,
```

Comment:
as far as I can tell, you can collapse all these jobs into a single one by using https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs; this would reduce massively the new changes (to a single job, with the same steps as any old one + matrix definition for env variables (`UPGRADE_BEFORE_DISABLE` x {`NODES`, `PORTS`, `EXT_STATUS`}))